### ISSUE TITLE
DL3025 does not support line wrap

### ISSUE THREAD

**Issue by kuwv** [hadolint/hadolint#953]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Some lines are too small to include within an cmd/entrypoint script while also being too long to have on one line. Wrapping an cmd/entrypoint causes a DL3025 error. This should not happen when escaping newlines.

### Actual behavior

```
DL3025 warning: Use arguments JSON notation for CMD and ENTRYPOINT arguments
```

### Steps to reproduce the behavior

Create an cmd/entrypoint and attempt to line wrap it with esacpe characters and perform a hadolint.

```
ENTRYPOINT [\
    "/some/command", \
    "/some/script", \
    "--some-arg=someval", \
]
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Additional environment details (OS, stack version, etc.)**

MacOS Ventura 13.2.1

**Comment by Eljees** [comment]
Line wrapping is supported — the escaped newlines are not what triggers this. The snippet has a trailing comma before `]`, which makes the array invalid JSON, so Docker falls back to shell form and DL3025 is right to fire.

Same two files on hadolint 2.15.1:

```dockerfile
# DL3025
ENTRYPOINT [\
    "/some/command", \
    "--some-arg=someval", \
]

# clean, exit 0
ENTRYPOINT [\
    "/some/command", \
    "--some-arg=someval" \
]
```

Building both on `FROM scratch` shows why:

```
$ docker inspect -f '{{json .Config.Entrypoint}}' with-comma
["/bin/sh","-c","[    \"/some/command\",     \"--some-arg=someval\", ]"]

$ docker inspect -f '{{json .Config.Entrypoint}}' without-comma
["/some/command","--some-arg=someval"]
```

With the trailing comma the whole bracket text is handed to `sh -c`, which is exactly the case DL3025 exists to catch. Dropping it gives real exec form and the warning goes away, wrapped across lines and all.

So there is nothing to fix here as far as I can tell, and this can probably be closed.
